zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/ProcessInstanceStateTransitionGuard.java

```diff
@@ -207,6 +209,22 @@ private Either<String, ElementInstance> hasNonInterruptedFlowScope(
     }
   }
 
+  private Either<String, ?> hasActiveProcessInstance(final BpmnElementContext context) {
+    if (context.getBpmnElementType() == BpmnElementType.PROCESS) {
+      return Either.right(null);
+    }
+    final var processInstance = stateBehavior.getElementInstance(context.getProcessInstanceKey());
+    if (processInstance.isInterrupted()
+        && !processInstance.getInterruptingElementId().equals(context.getElementId())) {
+      return Either.left(
+          String.format(
+              "Expected process instance to be not interrupted but was interrupted by an event with id '%s'.",
+              BufferUtil.bufferAsString(processInstance.getInterruptingElementId())));
```

Comment:
👍 Good to share the event element id in the failure message
🔧 We can remove the unnecessary negation (not interrupted -> active)

```suggestion
          String.format(
              "Expected process instance to be active but was interrupted by an event with id '%s'.",
              BufferUtil.bufferAsString(processInstance.getInterruptingElementId())));
```